bin/fin

```diff
@@ -5167,47 +5237,93 @@ addon_install ()
 		echo-red "Pre-install hook has failed and aborted the installation."
 		# Delete downloaded files if pre-install hook fails
 		[[ -d "$addon_path" ]] && [[ "$addon_path" =~ ".docksal" ]] &&
-			rm -r "$addon_path"
+		rm -r "$addon_path"
 		return 1
 	fi
 
-	# Download main script
-	echo-green "Downloading addon main script"
-	echo "  $addon_name/$addon_name"
-	curl -fsL "$URL_ADDON/$addon_name?$RANDOM" -o "$addon_path/$addon_name"
+	# Execute Main Script
 	chmod +x "$addon_path/$addon_name"
 	if_failed_error "Could not get $addon_name" "Check your internet connection and try again"
 
-	# Download other files
-	if [[ "$file_list" != "" ]]; then
-		# Exclude hooks
-		file_list=$(echo "$file_list" | grep -v "$hooks_regex")
-		if [[ "$file_list" != "" ]]; then
-			echo-green "Downloading other addon files..."
-			(
-				cd "$addon_path" || exit 1
-				for f in ${file_list}; do
-					echo "  $f"
-					# Filename can contain subdir name, which we need to create
-					local subdir=$(dirname "$f")
-					if [[ "$subdir" != "." ]]; then
-						mkdir -p "$subdir" || exit 1
-					fi
-					# Exit subshell with error if download failed
-					curl -fsSL "$URL_ADDON/$f?$RANDOM" -o "$f" || exit 1
-				done
-			)
-			if_failed_error "Download has failed" "Check log above for messages"
-		fi
-	fi
-
 	# Post-install hook
 	addon_hook "$addon_path" "$addon_name" "post-install" "$addon_global"
 	[[ $? != 0 ]] && _with_errors="${yellow}with errors${NC}"
 	[[ "$2" == "global" ]] && _globally="globally"
 	echo-green "Addon ${NC}$addon_name${green} was installed ${_globally} ${_with_errors}"
 }
 
+# Update addon
+# $1 - addon name
+# $2 - optional flag to install globally
+addon_update () {
```

Comment:
How is this function different from addon_install?

Reply:
@achekulaev `addon_update` only handles the updating piece. I believe the goal would be to eventually make it so that there were `pre-update` and `post-update` hooks for addons. Eventually also there could be the capability of versioning.

Reply:
I get that in future there might be the case, but is there any real code difference with `addon_install` right now? Because in PR review they look identical to me. And if they are identical right now I would just call `addon_install` for `fin addon update`

Reply:
the only difference would be the hooks. So seeing as the same code is ran for them both maybe refactoring the code and putting it in a function of it's own so it isn't duplicate coding.